lib/src/clojure_lsp/kondo.clj

```diff
@@ -162,30 +157,38 @@
 
 (defn ^:private normalize
   "Put kondo result in a standard format, with `analysis` normalized and
-  `analysis` and `findings` indexed by filename."
+  `analysis` and `findings` indexed by uri."
   [{:keys [analysis findings] :as kondo-results}
-   {:keys [external? ensure-filenames filter-analysis] :or {filter-analysis identity}}]
-  (let [analysis (->> analysis
+   {:keys [external? ensure-uris filter-analysis] :or {filter-analysis identity}}
+   db]
+  (let [filename->uri (memoize #(shared/filename->uri % db))
+        trade-filename-for-uri (fn [obj]
+                                 (-> obj
+                                     (assoc :uri (filename->uri (:filename obj)))
```

Comment:
Maybe we should assoc only if kondo already didn't return a URI for that obj?

Reply:
Maybe overhead for now, so it's ok

Reply:
I hope that kondo will start returning URI and stop returning filename, in which case this code can be removed entirely. But we can figure out those details when the time comes.